x-pack/plugins/ml/public/url_generator.test.ts

```diff
@@ -4,23 +4,233 @@
+            to: '2020-07-21T11:00:13.173Z',
+          },
+          query: {
+            analyze_wildcard: true,
+            query: '*',
+          },
+        });
+        expect(url).toBe(
+          "/app/ml/timeseriesexplorer?_g=(ml:(jobIds:!(logs_categorization_1)),refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2020-07-12T00:39:02.912Z',mode:absolute,to:'2020-07-22T15:52:18.613Z'))&_a=(mlTimeSeriesExplorer:(detectorIndex:0,entities:(mlcategory:'2')),query:(query_string:(analyze_wildcard:!t,query:'*')),zoom:(from:'2020-07-20T23:58:29.367Z',to:'2020-07-21T11:00:13.173Z'))"
+        );
+      });
+    });
+  });
+
+  describe('DataFrameAnalytics', () => {
+    describe('JobManagement Page', () => {
+      it('should generate valid URL for the Data Frame Analytics job management page', async () => {
+        const url = await urlGenerator.createUrl({
+          page: ML_PAGES.DATA_FRAME_ANALYTICS,
+        });
+        expect(url).toBe('/app/ml/data_frame_analytics');
+      });
+      it('should generate valid URL for the Data Frame Analytics job management page with jobId', async () => {
+        const url = await urlGenerator.createUrl({
+          page: ML_PAGES.DATA_FRAME_ANALYTICS,
+          jobId: 'grid_regression_1',
+        });
+        expect(url).toBe('/app/ml/data_frame_analytics?mlManagement=(jobId:grid_regression_1)');
+      });
+
+      it('should generate valid URL for the Data Frame Analytics job management page with groupIds', async () => {
+        const url = await urlGenerator.createUrl({
+          page: ML_PAGES.DATA_FRAME_ANALYTICS,
+          groupIds: ['group_1', 'group_2'],
+        });
+        expect(url).toBe('/app/ml/data_frame_analytics?mlManagement=(groupIds:!(group_1,group_2))');
+      });
+    });
+
+    describe('ExplorationPage', () => {
+      it('should generate valid URL for the Data Frame Analytics exploration page for job', async () => {
+        const url = await urlGenerator.createUrl({
+          page: ML_PAGES.DATA_FRAME_ANALYTICS_EXPLORATION,
+          jobId: 'grid_regression_1',
+          analysisType: ANALYSIS_CONFIG_TYPE.REGRESSION,
+        });
+        expect(url).toBe(
+          '/app/ml/data_frame_analytics/exploration?_g=(ml:(analysisType:regression,jobId:grid_regression_1))'
+        );
+      });
+    });
+  });
+
+  describe('DataVisualizer', () => {
+    it('should generate valid URL for the Data Visualizer page', async () => {
+      const url = await urlGenerator.createUrl({
+        page: ML_PAGES.DATA_VISUALIZER,
+      });
+      expect(url).toBe('/app/ml/datavisualizer');
+    });
+
+    it('should generate valid URL for the Data Visualizer import file page', async () => {
+      const url = await urlGenerator.createUrl({
+        page: ML_PAGES.DATA_VISUALIZER_FILE,
+      });
+      expect(url).toBe('/app/ml/filedatavisualizer');
+    });
+
+    it('should generate valid URL for the Data Visualizer import by index page', async () => {
```

Comment:
Change description to `should generate valid URL for the Index Data Visualizer select index pattern or saved search page`

Reply:
Updated here https://github.com/elastic/kibana/pull/75696/commits/3a1a7cba1df24ca501c0a7061bff2fc733688f33